keyboards/splitkb/elora/rev1/rev1.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2024 splitkb.com (support@splitkb.com)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+#include "rev1.h"
+
+#include "spi_master.h"
+#include "matrix.h"
+#include "keyboard.h"
+
+#include "myriad.h"
+
+bool is_oled_enabled = true;
+
+//// HW init
+
+// Make sure all external hardware is
+// in a known-good state on powerup
+void keyboard_pre_init_kb(void) {
+    /// SPI Chip Select pins for various hardware
+    // Matrix CS
+    gpio_set_pin_output(GP13);
+    gpio_write_pin_high(GP13);
+    // Myriad Module CS
+    gpio_set_pin_output(GP9);
+    gpio_write_pin_high(GP9);
+
+    gpio_set_pin_output(ELORA_CC1_PIN);
+    gpio_write_pin_low(ELORA_CC1_PIN);
+
+    gpio_set_pin_output(ELORA_CC2_PIN);
+    gpio_write_pin_low(ELORA_CC2_PIN);
+
+    // We have to get the SPI interface working quite early,
+    // So make sure it is available well before we need it
+    spi_init();
+    
+    keyboard_pre_init_user();
+}
+
+void keyboard_post_init_kb(void) {
+
+    keyboard_post_init_user();
+}
+
+//// Matrix functionality
```

Comment:
It would be preferable for the custom matrix implementation to live in a `matrix.c` file. ([ref](https://docs.qmk.fm/custom_matrix#prerequisites))